### ISSUE TITLE
MAML: moving mean and variance (of batchnorm layers) not updated during training

### ISSUE THREAD

**Issue by keshik6** [siavash-khodadadeh/MetaLearning-TF2.0#5]
Hi, thanks for the code. 

This issue is wrt to the MAML code ( MetaLearning-TF2.0/models/maml/maml.py )

I found an issue with moving mean and variance of batchnorm layers. They are not updated during training. I believe this is due to using custom keras training loop. 

Keras automatically handles this when using model.fit(), but when training using custom keras loop I believe it has to be handled separately. 

In TF 1.x, this is usually done by adding the update_ops dependencies as shown below.

```
update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)
    with tf.control_dependencies(update_ops):
      train_op = optimizer.minimize(loss)
```

But not sure as to how this can be handled in TF 2.x.

Please share your thoughts on this.

Thanks!

**Comment by siavash-khodadadeh** [comment]
Hello, this is intentional because we use the momentum of 0.0 in our batch norm layers which means for each task these statistics are evaluated based on just train set of that particular task.
Look at here for example https://github.com/siavash-khodadadeh/MetaLearning-TF2.0/blob/master/networks.py#L14

We will release a much better of this repository soon in which we enable both options to update or not to update.

In TF2, you can handle updating of these parameters by setting training=True or False when calling these layers. However, our current implementation does not propagate batch norm statistics so they do not update.

**Comment by keshik6** [comment]
Alright!

Thank you for clarifying.
